src/backend/storeManagers/gog/games.ts

```diff
@@ -699,69 +835,208 @@ export async function uninstall({ appName }: RemoveArgs): Promise<ExecResult> {
 
     const command = [
       uninstallerPath,
-      '/verysilent',
-      `/dir=${shlex.quote(installDirectory)}`
+      '/VERYSILENT',
+      `/ProductId=${appName}`,
+      '/galaxyclient',
+      '/KEEPSAVES'
     ]
 
     logInfo(['Executing uninstall command', command.join(' ')], LogPrefix.Gog)
 
     if (!isWindows) {
-      await runWineCommandUtil({
-        gameSettings,
-        commandParts: command,
-        wait: true,
-        protonVerb: 'waitforexitandrun'
-      })
+      if (existsSync(gameSettings.winePrefix)) {
+        await runWineCommandUtil({
+          gameSettings,
+          commandParts: command,
+          wait: true
+        })
+      }
     } else {
       const adminCommand = [
         'Start-Process',
         '-FilePath',
         uninstallerPath,
         '-Verb',
         'RunAs',
+        '-Wait',
         '-ArgumentList'
       ]
 
       await spawnAsync('powershell', [
         ...adminCommand,
-        `/verysilent /dir=${shlex.quote(installDirectory)}`
+        `"/verysilent","\`"/dir=${installDirectory}\`""`,
+        ``
       ])
     }
-  } else {
-    if (existsSync(object.install_path))
-      rmSync(object.install_path, { recursive: true })
+  }
+  if (existsSync(object.install_path)) {
+    rmSync(object.install_path, { recursive: true })
+  }
+  const manifestPath = join(gogdlConfigPath, 'manifests', appName)
+  if (existsSync(manifestPath)) {
+    rmSync(manifestPath) // Delete manifest so gogdl won't try to patch the not installed game
+  }
+  const supportPath = join(gogSupportPath, appName)
+  if (existsSync(supportPath)) {
+    rmSync(supportPath, { recursive: true }) // Remove unnecessary support dir
   }
   installedGamesStore.set('installed', array)
   refreshInstalled()
   const gameInfo = getGameInfo(appName)
+  gameInfo.is_installed = false
+  gameInfo.install = { is_dlc: false }
   await removeShortcutsUtil(gameInfo)
   syncStore.delete(appName)
   await removeNonSteamGame({ gameInfo })
-  sendFrontendMessage('refreshLibrary', 'gog')
+  sendFrontendMessage('pushGameToLibrary', gameInfo)
```

Comment:
Since gog integration is so tied up to Heroic I feel like we should move towards making less full library refreshes